src/main/java/com/powsybl/openloadflow/sa/WoodburyDcSecurityAnalysis.java

```diff
@@ -125,11 +148,21 @@ private double[] calculatePostContingencyStatesForAContingency(DcLoadFlowContext
     }
 
     /**
-     * Calculate post contingency states for a contingency breaking connectivity.
+     * Calculate post contingency states for a contingency breaking connectivity, without remedial actions.
      */
     private double[] calculatePostContingencyStatesForAContingencyBreakingConnectivity(ConnectivityBreakAnalysis.ConnectivityAnalysisResult connectivityAnalysisResult, DcLoadFlowContext loadFlowContext,
-                                                                                      Map<String, ComputedContingencyElement> contingencyElementByBranch, double[] flowStates, DenseMatrix contingenciesStates,
-                                                                                      ReportNode reportNode) {
+                                                                                       Map<String, ComputedContingencyElement> contingencyElementByBranch, double[] flowStates, DenseMatrix contingenciesStates,
+                                                                                       ReportNode reportNode) {
+        return calculatePostContingencyAndOperatorStrategyStatesForAContingencyBreakingConnectivity(connectivityAnalysisResult, loadFlowContext, contingencyElementByBranch, flowStates,
+                contingenciesStates, Collections.emptyList(), Collections.emptyMap(), WoodburyEngine.EmptyDenseMatrix.getInstance(), reportNode);
+    }
+
+    /**
+     * Calculate post contingency and post operator strategy states, for a contingency breaking connectivity.
+     */
+    private double[] calculatePostContingencyAndOperatorStrategyStatesForAContingencyBreakingConnectivity(ConnectivityBreakAnalysis.ConnectivityAnalysisResult connectivityAnalysisResult, DcLoadFlowContext loadFlowContext,
```

Comment:
The code calling this function has those comments
                                // no need to distribute active mismatch due to connectivity modifications
                                // this is handled when the slack is distributed in pre contingency states override

I see nothing in the method that handles distributed slack among connex components.
How dos this behave if a contingency breaks connectivity and you end up with two parts where in **each** part
sum of loads != sum of production ? Do you have a test for that N

Reply:
This method will call `calculatePostContingencyAndOperatorStrategyStatesForAContingency`, which will call `DcLoadFlowEngine.run` (because the list of disabled buses is not empty, as there is a connectivity break). In this method, slack is distributed **on buses that are not disabled**. 

So if a contingency breaks connectivity and there is two parts where in each part sum of loads != sum of production, only the active slack of main connex component (including slack bus) will be distributed among buses of that component. 

The already existing TU `testConnectivityResultOnSplitDc` illustrates this behavior. Both components are unbalanced after the contingency, but slack is distributed correctly **in fast DC SA**.

Reply:
I gave a closer look and this is what happens:
    If a contingency breaks a network in two parts,  all bus of the second part are disabled.
   So the broken part is not computed, even in fast DC. So there's no issue if there is no slack bus on this part.

   This being said, I wonder what happens if the slack bus is on the disconnected part of the network because relocateSlackBus is passed to false int toLfContingency.

Reply:
I see that you have added a note in the doc related to this:
<pre>
--- doc extract
- Slack relocation following the application of a contingency is not supported. 
As a result, security analysis is carried out only in slack component, and not necessarily in the larger one.
</pre>

Do you have a test that shows the behaviour when the slack but is not in the main component after a contingency breaking connectivity ? That documented behaviour didn't seem clear to me from the code - I didn't find anything related to that in toLfContngency (the main component doesn't seem to be changed if relocateSlackBus is false ans the slack bus is isolated)

Reply:
After stepping into the code, the documented behaviour might be true but:
   - 1) There is no test the verifies this. Can you add one ? 
   - 2) This seems to be due by a side effect of network.getConnectvity called at the beginning of PropagaetdContingency.findBusesAndBranchesImpactedBecauseOfConnectivityLoss (it sets the main component to the one containing the slack bus). Can you add a comment before the call to connectivity = network.getConnectivity(); to explain that this redefines the main compoment to the one containinge the first slack bus ? That would make the code easier to understand.

Reply:
I added the comment, and the TU `testDcFastModeInFirstSlackComponentOnly` to clarify this behaviour. Thanks for the careful review @vidaldid-rte.